Sources/AuthFoundation/OAuth2/OAuth2Client.swift

```diff
@@ -96,8 +107,11 @@ public final class OAuth2Client {
         
         self.configuration = configuration
         self.session = session ?? URLSession(configuration: .ephemeral)
-        
-        NotificationCenter.default.post(name: .oauth2ClientCreated, object: self)
+        self.refreshQueue = DispatchQueue(label: "com.okta.refreshQueue.\(configuration.issuerURL.host ?? "unknown")",
+                                          qos: .userInitiated,
+                                          attributes: .concurrent)
+
+        TaskData.notificationCenter.post(name: .oauth2ClientCreated, object: self)
```

Comment:
That pattern has some pitfalls as we send notification with `self` for a partially constructed/initialized object. Since notification center calls all callbacks synchronously it means that all callbacks will be executed within `init` context. There is no guarantee that observers will not start using `self` immediately(which is still being constructed)

Reply:
The initializer has already completed at this point, and is wiring up its communication with the credential coordinator.  Fortunately the compiler won't let `self` be used until all its properties have been initialized, so at this point it's safe for notification observers to use the object.

The reason why it's important is that there are many occasions where an OAuth2Client may be implicitly created on behalf of another actor in the system (e.g. an AuthenticationFlow), and it could be important for a delegate to observe events / update the object before it's used.

Reply:
It still looks weird to me that we postpone construction of the object with blocking notification. That means that `let obj = Object()` call will be postponed until all subscribers finish their processing(and we have no idea how heavy it can be)
I suggest either fire this notification in async task or use NotificationQueue instead

Reply:
I feel like dispatching this async is the correct approach, but I'd need better testing there to ensure the appropriate consumers get notified in a timely fashion.  Or, even better, the internal consumers of this notification that need guaranteed access to the object before it's used downstream, should have a private way of being notified synchronously.

I'll look into this ...